### PR TITLE
Introduce AttributeFilters' support for value filtering

### LINKED ISSUE #722: Support value filtering by AFM attribute filters

### PR DESCRIPTION
**Pull request by jimirocks**
resolves #722

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/CompatibilityAttributeFilterElements.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+import java.io.IOException;
+import java.util.List;
+
+import static com.fasterxml.jackson.databind.JsonMappingException.from;
+
+@Deprecated
```
Comment (liry) [inline gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/CompatibilityAttributeFilterElements.java]: plz add some javadoc, why it's here and when can we remove it

File: gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/CompatibilityAttributeFilterElements.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+public interface CompatibilityAttributeFilterElements {
```
Comment (liry) [inline gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/CompatibilityAttributeFilterElements.java]: Is this interface used anywhere in the end?

File: gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/AttributeFilterElements.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import static com.fasterxml.jackson.databind.JsonMappingException.from;
+
+/**
+ * Represents elements, which can be used as value of "in" or "notIn" by {@link AttributeFilter}.
+ */
+@JsonSerialize(using = AttributeFilterElements.Serializer.class)
+@JsonDeserialize(using = AttributeFilterElements.Deserializer.class)
+public interface AttributeFilterElements {
+
+    List<String> getElements();
```
Comment (liry) [inline gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/AttributeFilterElements.java]: Would be nice to have some javadoc here.

File: gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/CompatibilityAttributeFilterElements.java (hunk 1 above)
Comment (jimirocks) [reply gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/CompatibilityAttributeFilterElements.java]: ups, this class is not needed in final solution - will delete it

File: gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/CompatibilityAttributeFilterElements.java (hunk 2 above)
Comment (jimirocks) [reply gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/CompatibilityAttributeFilterElements.java]: no ...

**Comment by peter-plochan** [comment]
please fix @liry comments, otherwise ok 👍

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/AttributeFilterElements.java (hunk 3 above)
Comment (jimirocks) [reply gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/AttributeFilterElements.java]: javadoc written

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-java-merge-compliance-check/52/ : SUCCESS in 19s

**Pull request merged**